serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc

```diff
@@ -0,0 +1,141 @@
+to perform HTTP requests,  the resource defined in the URI will then be queried in the current Kubernetes cluster and
+translated to a valid URL. The service discovery feature works during the Application startup, it scans all Quarkus
+configuration in search of the following URI pattern, keep in mind that, if the startup time matter, consider to use
+static URL instead.
+The following sketch demonstrates how this URI is composed:
+
+[source,shell]
+----
+kubernetes:<group>/<version>/<kind>/<namespace>/<name>?<attributeName>=<attributeValue>
+\________/ \_____/ \_______/ \____/ \_________/ \____/ \______________________________/
+  scheme    group   version   kind   namespace   name   additional resource attributes
+            \____________________/                       \__________________________/
+                Kubernetes GVK                            Supported value: port_name
+----
+
+The following scheme values are supported:
+
+* **kubernetes**
+* **openshift**
+* **knative**
+
+
+For the GVK, the following Kubernetes Resources are currently supported:
+
+* **v1/service**
+* **serving.knative.dev/v1/service**
+* **v1/pod**
+* **apps/v1/deployment**
+* **apps.openshift.io/v1/deploymentconfig**
+* **apps/v1/statefulset**
+* **route.openshift.io/v1/route**
+* **networking.k8s.io/v1/ingress**
+
+[IMPORTANT]
+====
+The **GVK** (Group, Version and Kind) and the **Name** values are **mandatory** and can't be empty, otherwise, the engine will throw a validation issue and the discovery will not be performed.  +
+The **namespace** is **optional**. In case of an empty value, the current namespace or context will be used.
+====
+
+== Configuration
+
+You can configure the Service Discovery with two **build time** properties, which are:
+
+* `quarkus.kogito.sw.discovery.enabled`: Enable the service discovery. Defaults to _true_.
+* `quarkus.kogito.kubernetes-client.okhttp.static.logger`: Weather or not to print the `http` communication between the client and the Kubernetes API. Defaults to _false_.
+
+
+== How does it work?
+
+During application startup, the service discovery is performed at the Quarkus _STATIC_INIT_ time. First, it scans +
+the Quarkus configuration values and searches for the Kubernetes URI pattern. Then if at least one is found, the engine +
+will parse the URI, query the Kubernetes API for the given resource, and override the given system property.
+
+Suppose we have an application that will consume a resource running on Kubernetes, this resource is a Knative Service +
+that exposes a function that can be discovered with the following URI:
+
+
+[source,shell]
+----
+org.kie.kogito.sw.knative.service=knative:v1/Service/serverless-workflow-greeting-quarkus/greeting-quarkus-cli
+----
+
+[NOTE]
+====
+Note that, the application property does not matter. The Service Discovery only cares about the application property value.
```

Comment:
Honestly, I don't get that sentence. The application property does not matter but the Service Discovery only cares about it ?

Reply:
indeed, needs rephrase.
the application property name does not matter, only its value.